contrib/demo/ingress-startKcp.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+CURRENT_DIR="$(pwd)"
+DEMO_ROOT="$(dirname "${BASH_SOURCE}")"
+KCP_ROOT="$(cd ${DEMO_ROOT}/../.. && pwd)"
+TEMP_DIR="$(mktemp -d)"
+CLUSTERS_DIR="${DEMO_ROOT}/clusters"
+export KCP_DATA_ROOT=${KCP_DATA_ROOT:-$KCP_ROOT}
+
+source "${DEMO_ROOT}"/.startUtils
+setupTraps $0
+
+if ! command -v envoy &> /dev/null
```

Comment:
Move this up to the top? That way we wouldn't create a temp dir if envoy isn't present.